staging/src/k8s.io/cloud-provider/app/controllermanager.go

```diff
@@ -320,48 +320,80 @@ var ControllersDisabledByDefault = sets.NewString()
 
 // ConstructControllerInitializers is a private map of named controller groups (you can start more than one in an init func)
 // paired to their InitFunc.  This allows for structured downstream composition and subdivision.
-func ConstructControllerInitializers(initFuncConstructors map[string]InitFuncConstructor, completedConfig *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) map[string]InitFunc {
+func ConstructControllerInitializers(controllerInitializerConstructors map[string]ControllerInitializerConstructor, completedConfig *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) map[string]InitFunc {
 	controllers := map[string]InitFunc{}
-	for name, constructor := range initFuncConstructors {
-		controllers[name] = constructor(completedConfig, cloud)
+	for name, constructor := range controllerInitializerConstructors {
+		controllers[name] = constructor.Constructor(constructor.InitContext, completedConfig, cloud)
 	}
 	return controllers
 }
 
+type ControllerInitializerConstructor struct {
```

Comment:
not huge fan of this name, but I have no better suggestion. Initializer and constructor, what is the difference?  also there will be ControllerInitializerConstructor.Constructor. 

InitFuncConstructor kinda makes sense because it is a function (constructor) that returns an InitFunc.

Reply:
I'm open to suggestions on naming.  InitFuncConstructor is fine with me, I think ControllerInitializerConstructor is similar but ControllerInitializer is a better descriptor of what the function does than InitFunc.  But with InitFunc the name matches what its constructing.